gradio/image_utils.py

```diff
@@ -26,9 +27,15 @@ def format_image(
     elif type == "numpy":
         return np.array(im)
     elif type == "filepath":
-        path = processing_utils.save_pil_to_cache(
-            im, cache_dir=cache_dir, name=name, format=fmt or "png"  # type: ignore
-        )
+        try:
+            path = processing_utils.save_pil_to_cache(
+                im, cache_dir=cache_dir, name=name, format=fmt or format  # type: ignore
+            )
+        # Catch error if format is not supported by PIL
+        except Exception:
```

Comment:
looks like its a KeyError that PIL raises for invalid formats
```suggestion
        except KeyError:
```

Reply:
Is that the only one?

Reply:
For invalid formats, yes. Tested with:

```py
from PIL import Image

Image.open("cheetah.jpg").save("c2.jpg", format="cur")  # cur is a read only format
```

Reply:
Let's see what happens!